jdbc/src/java/org/apache/hive/jdbc/HiveConnection.java

```diff
@@ -1002,11 +1002,19 @@ private String getSessionValue(String varName, String varDefault) {
     return varValue;
   }
 
-  // copy loginTimeout from driver manager. Thrift timeout needs to be in millis
+  // use socketTimeout from jdbc connection url. Thrift timeout needs to be in millis
   private void setupLoginTimeout() {
-    long timeOut = TimeUnit.SECONDS.toMillis(DriverManager.getLoginTimeout());
+    String socketTimeoutStr = sessConfMap.getOrDefault(JdbcConnectionParams.SOCKET_TIMEOUT, "0");
+    long timeOut = 0;
```

Comment:
What if the default value set to TimeUnit.SECONDS.toMillis(DriverManager.getLoginTimeout())?

Reply:
This issue describes the problem: [HIVE-22196](https://issues.apache.org/jira/browse/HIVE-22196) : Socket timeouts happen when other drivers set DriverManager.loginTimeout.

Reply:
Set the default value to 0 may cause problems for those elder clients using DriverManager.loginTimeout to prevent the client from hang or return quickly if there are some problems in HS2, users have to refactor the uri with the socket timeout parameter after applying this.  After all, there are some preconditions should be met on the HIVE-22196: 1, There are different drivers that created in the same jvm; 2,  One of the driver sets the loginTimeout that is inappropriate for others.

Reply:
Sorry for the late reply. If users depends on DriverManager.loginTimeout to control HS2 jdbc timeout, they will have to refactor their code after applying this. However, we haven't yet met a situation that we need this timeout to return quickly when HS2 hangs or something. Our problem is when submitting a big query which needs minutes to compile, the connection is lost because of this DriverManager.loginTimeout. Maybe we need to set a timeout parameter for HS2 on the server side too.